### PR TITLE
Support `pin_memory()` in `Multi{Embedding,Nested}Tensor` and `TensorFrame`

### PR DESCRIPTION
**Pull request by akihironitta**

### CODE REVIEW

**Review by yiweny — COMMENTED** [review]
What is `pin_memory`? 🙂‍↔️

**Review by akihironitta — COMMENTED** [review]
It's useful for moving data to GPU faster: https://pytorch.org/docs/stable/data.html#memory-pinning :)

File: torch_frame/data/multi_tensor.py [hunk 1]
```diff
@@ -93,6 +93,12 @@ def cpu(self, *args, **kwargs):
     def cuda(self, *args, **kwargs):
         return self._apply(lambda x: x.cuda(*args, **kwargs))
 
+    def pin_memory(self, *args, **kwargs):
+        return self._apply(lambda x: x.pin_memory(*args, **kwargs))
+
+    def is_pinned(self) -> bool:
+        return self.values.is_pinned()
```
Comment (zechengz) [inline torch_frame/data/multi_tensor.py:100]: Should this be
```suggestion
        return self.values.is_pinned() and self.offset.is_pinned()
```

**Review by zechengz — APPROVED** [review]
LGTM.
IMO usually we set pin_memory for the dataloader right?

**Review by akihironitta — COMMENTED** [review]
> IMO usually we set pin_memory for the dataloader right?

Yes! The dataloader will now call this method on our multi tensors and tensor frame.

**Comment by weihua916** [comment]
are we merging this? Why there are a lot of test failures?

**Comment by akihironitta** [comment]
Will be merging this PR once the CI becomes green 🟢 Let me know if anyone has any review comments on this PR as I'm happy to address any concerns or elaborate on the change in this PR :)

File: test/data/test_multi_nested_tensor.py [hunk 2]
```diff
@@ -95,8 +95,8 @@ def test_fillna_col():
                           torch.tensor([100], dtype=torch.float32)))
 
 
-@withCUDA
-def test_multi_nested_tensor_basics(device):
+@onlyCUDA
+def test_basics(device):
```
Comment (akihironitta) [inline test/data/test_multi_nested_tensor.py:99]: #476

**Pull request merged**